tests/cases/conformance/internalModules/DeclarationMerging/ClassAndModuleWithSameNameAndCommonRootES6.ts

```diff
@@ -0,0 +1,39 @@
+// @target: ES6
+// @filename: class.ts
+module X.Y {
+    export class Point {
+        constructor(x: number, y: number) {
+            this.x = x;
+            this.y = y;
+        }
+        x: number;
+        y: number;
+    }
+}
+
+// @filename: module.ts
+module X.Y {
+    export module Point {
+        export var Origin = new Point(0, 0);
+    }
+}
+
+// @filename: test.ts
+//var cl: { x: number; y: number; }
+var cl = new X.Y.Point(1,1);
+var cl = X.Y.Point.Origin; // error not expected here same as bug 83996 ?
+
+
+// @filename: simple.ts

```

Comment:
This is just a copy of `ClassAndModuleWithSameNameAndCommonRoot.ts` that has its target set for ES6. I can simplify it (remove the cases that should follow the same code path as the ES3 version) but I don't think I'd split it into multiple files.